crates/astria-core/src/sequencer/v1/block/mod.rs

```diff
@@ -1239,6 +1241,7 @@ impl FilteredSequencerBlockError {
 /// A [`Deposit`] is constructed whenever a [`BridgeLockAction`] is executed
 /// and stored as part of the block's events.
 #[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize))]
```

Comment:
This should serialize exactly the same way as the protobuf. Use:
```rust
#[cfg_attr(feature = "serde", derive(serde::Serialize))]
#[cfg_attr(feature = "serde", serde(into = "::core::generated::sequencer::v1::Deposit"))]
```
Also update the compiler tool to generate code for the `Deposity` type and add a `impl From<Deposit> for ::core::generated::sequencer::v1::Deposit`